x/gov/genesis.go

```diff
@@ -60,6 +61,34 @@ func DefaultGenesisState() GenesisState {
 
 // ValidateGenesis TODO https://github.com/cosmos/cosmos-sdk/issues/3007
 func ValidateGenesis(data GenesisState) error {
+	threshold := data.TallyParams.Threshold
+	if threshold.IsNegative() || threshold.GT(sdk.OneDec()) {
+		return fmt.Errorf("Governance vote threshold should be positive and less than one, is %s",
+			threshold.String())
+	}
+
+	veto := data.TallyParams.Veto
+	if veto.IsNegative() || veto.GT(sdk.OneDec()) {
+		return fmt.Errorf("Governance vote veto threshold should be positive and less than one, is %s",
+			veto.String())
+	}
+
+	govPenalty := data.TallyParams.GovernancePenalty
+	if govPenalty.IsNegative() || govPenalty.GT(sdk.OneDec()) {
+		return fmt.Errorf("Governance vote veto threshold should be positive and less than one, is %s",
+			govPenalty.String())
+	}
+
+	if data.DepositParams.MaxDepositPeriod > data.VotingParams.VotingPeriod {
+		return fmt.Errorf("Governance deposit period should be less than the voting period (%ds), is %ds",
```

Comment:
This tests less than or equal to, did we want to test strictly less than?

Reply:
That tests `>` (less than) right?

Reply:
`>` is greater than, so this will succeed when `depositPeriod <= votingPeriod`

Reply:
Well 🤦‍♂️ thank you for the schooling.